### ISSUE TITLE
Accessing a model property that is a types.reference not working anymore (v3.3.0 to v3.10.1)

### ISSUE THREAD

**Issue by fbeaulieu** [mobxjs/mobx-state-tree#1164]
Hi there,

I tried to find an explanation from the change log and other opened issues/questions but no success there.
I updated MST from v3.3.0 to v3.10.1 and now, whenever I try to access (read) properties of types.reference, 
I get this error:

**Error: [mobx-state-tree] Failed to resolve reference 'my_ref_identifier' to type 'my_ref_model' (from node: /my_ref_model_property)**

Here is my testing code...
```
const ChildLevelModel = types
  .model({
    id: types.identifier,
    levelValue: '',
    dummyValue: ''
  })
  .named('ChildLevelModel')
  ;


const ParentLevelModel = types
  .model({
    levelValue: '',
    childLevelRef: types.maybe(types.reference(ChildLevelModel))
  })
  .actions(self => {
    return {
      setChildLevelRef: (value) => {
        self.childLevelRef = value;
      }
    }
  })
  .named('ParentLevelModel')
  ;

// Let's create some model instances...
var childLevelModelInstance = ChildLevelModel.create({ id: '1' });
var parentLevelModelInstance = ParentLevelModel.create();

// Setting my referenced property
parentLevelModelInstance.setChildLevelRef(childLevelModelInstance);

console.log(parentLevelModelInstance.childLevelRef); // ERROR


**Error: [mobx-state-tree] Failed to resolve reference '1' to type 'ChildLevelModel' (from node: /childLevelRef)**

```

Running the same code with MST v3.3.0 works perfectly. I know there have been some updates regarding references, but did I miss something obvious here and I was just lucky that it worked before updating to the latest release ?

_PS: I tried with v3.10.1 and with v3.10.2 as well and same behaviour..._

Thanx !

**Comment by xaviergonz** [comment]
The reference and referenced type must exist under the same root store. There used to be a bug that didn't check this. Custom references can be used to opt out of this behavior

**Comment by fbeaulieu** [comment]
> The reference and referenced instance must exist under the same root store. There used to be a bug that didn't check this. Custom references can be used to opt out of this behavior

Ok I see, and I guess by Custom references, you refer to the fact that we can override the references resolve logic right ?

**Comment by xaviergonz** [comment]
Yep! https://github.com/mobxjs/mobx-state-tree#customizable-references
